src/ln/channelmonitor.rs

```diff
@@ -1340,31 +1457,32 @@ impl ChannelMonitor {
 					htlc_success_tx.input[0].witness.push(our_sig.serialize_der(&self.secp_ctx).to_vec());
 					htlc_success_tx.input[0].witness[2].push(SigHashType::All as u8);
 
-					htlc_success_tx.input[0].witness.push(payment_preimage.to_vec());
+					htlc_success_tx.input[0].witness.push(payment_preimage.0.to_vec());
 					htlc_success_tx.input[0].witness.push(chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key).into_bytes());
 
 					add_dynamic_output!(htlc_success_tx, 0);
 					res.push(htlc_success_tx);
 				}
+			watch_outputs.push(local_tx.tx.output[htlc.transaction_output_index as usize].clone());
```

Comment:
Hmm test didn't catch this because it wasn't covered explicitly by them, but correcting that broke test_static_spendable_outputs_justice_tx_revoked_htlc_timeout_tx and test_dynamic_spendable_outputs_local_htlc_timeout_tx, due to block rescan caused by detection extension